### DISCUSSION TITLE
How to Limit a 3D Model's Distance in AR

### DISCUSSION THREAD

**Discussion by ProduktiaDev** [google/model-viewer#4967] [Q&A]
Hello, I’m trying to make the 3D model projected in augmented reality stay within 2 meters from the center of the camera, but I can’t figure out how to do it. Does anyone know how?

**Comment by elalish** [comment]
Generally in AR, the object is locked into world-space so it looks real. That's how we do it anyway. Can you show a video of what you're hoping to achieve?

**Reply by ProduktiaDev** [reply]
As I show you in the video, I want the maximum distance the 3D moves away to be about 1 meter from the center of the camera, but right now it moves about 3 or 4 meters away.
 

https://github.com/user-attachments/assets/a6dd86a0-e947-438c-962b-5a56e2b8a99a

**Reply by elalish** [reply]
What do you want to happen if the model is placed correctly and the user walks away from it? Should it follow them? If so, how? You're welcome to make a PR to improve our UX. We don't currently have an API for what you're asking for.

**Reply by ProduktiaDev** [reply]
What I want is to prevent the user from placing the AR object more than 2 meters away from the camera's position. In other words, I want to limit the maximum distance at which the object can be positioned, as the user can currently place it around 4-5 meters away.
![image](https://github.com/user-attachments/assets/a6667840-6207-4a85-badb-afec7bb3b1b9)

**Reply by elalish** [reply]
PRs welcome, but as I say, even if you restrict their initial placement distance, they can move themselves further away. Then when they try to move it again, and it's already out of range, what should happen? A sudden jump? Or they can only bring it closer, but not further away again? There are a lot of subtle UX decisions to be made in what you're describing.

**Reply by ProduktiaDev** [reply]
Okay, I understand what you mean. Let me explain it better now:

What I want is to implement a restriction on how far the 3D object can be moved from its initial position, limiting it to a maximum of 2 meters. However, if the user moves closer to the 3D object, it should be possible to move it another 2 meters from the user’s new position.

The idea is to keep everything else working as it does currently. For example, if you move the object 2 meters away and then you step back another 2 meters, the AR object would remain 4 meters away, just as it works by default now. Additionally, if the object is too far and you can't interact with it, you need to either move closer or restart the AR to have the object nearby again.

I think there should be a parameter in the code like maxDistanceMove or something similar to handle this, but I’m not sure if it’s already implemented.

Thanks.

**Reply by elalish** [reply]
It's not, and it sounds a little bit complicated, but if you're willing to make the PR, I'll happily review it.

**Reply by ProduktiaDev** [reply]
Okay, thanks for your help
